File: src/main/java/com/dambra/paul/stringcalculator/StringCalculator.java
package com.dambra.paul.stringcalculator;

class StringCalculator {

    String calculate(String input) {
        var stringOperations = SplitToStringOperations.from(input);
        var parts = RecursivelyProcessBracketedCalculations.from(stringOperations);
        var calculationStart = StringOperationsToCalculationParts.from(parts);
        var result = calculationStart.Calculate(0);
        return RemoveTrailingPointZero.from(result);
    }

}

File: src/main/java/com/dambra/paul/stringcalculator/StringCalculator.kt
package com.dambra.paul.stringcalculator

internal class StringCalculator {

    fun calculate(input: String): String {
        val stringOperations = SplitToStringOperations.from(input)
        val parts = RecursivelyProcessBracketedCalculations.from(stringOperations)
        val calculationStart = StringOperationsToCalculationParts.from(parts)
        val result = calculationStart.calculate(0.0)
        return result.toString().removeSuffix(".0")
    }

}
